include/net/lorawan.h

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2020 Manivannan Sadhasivam <mani@kernel.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_LORAWAN_H_
+#define ZEPHYR_INCLUDE_NET_LORAWAN_H_
+
+/**
+ * @file
+ * @brief Public LoRaWAN APIs
+ */
+
+#include <device.h>
+
+/**
+ * @brief LoRaWAN class types.
+ */
+enum lorawan_class {
+	LORAWAN_CLASS_A = 0x00,
+	LORAWAN_CLASS_B = 0x01,
+	LORAWAN_CLASS_C = 0x02,
+};
+
+/**
+ * @brief LoRaWAN activation types.
+ */
+enum lorawan_act_type {
+	LORAWAN_ACT_OTAA = 0,
+	LORAWAN_ACT_ABP,
+};
+
+/**
+ * @brief LoRaWAN datarate types.
+ */
+enum lorawan_datarate {
+	LORAWAN_DR_0 = 0,
+	LORAWAN_DR_1,
+	LORAWAN_DR_2,
+	LORAWAN_DR_3,
+	LORAWAN_DR_4,
+	LORAWAN_DR_5,
+	LORAWAN_DR_6,
+	LORAWAN_DR_7,
+	LORAWAN_DR_8,
+	LORAWAN_DR_9,
+	LORAWAN_DR_10,
+	LORAWAN_DR_11,
+	LORAWAN_DR_12,
+	LORAWAN_DR_13,
+	LORAWAN_DR_14,
+	LORAWAN_DR_15,
+};
+
+/**
+ * @brief LoRaWAN message types.
+ *
+ * Note: The default behaviour is LORAWAN_MSG_UNCONFIRMED
```

Comment:
This is refers to an enum entry that doesn't exist.

Reply:
Fixed!